src/rdkafka_offset.c

```diff
@@ -413,16 +413,22 @@ rd_kafka_commit (rd_kafka_t *rk,
         rd_kafka_cgrp_t *rkcg;
 	rd_kafka_resp_err_t err;
 	rd_kafka_q_t *repq = NULL;
+	rd_kafka_replyq_t rq = RD_KAFKA_NO_REPLYQ;
 
         if (!(rkcg = rd_kafka_cgrp_get(rk)))
                 return RD_KAFKA_RESP_ERR__UNKNOWN_GROUP;
 
         if (!async)
                 repq = rd_kafka_q_new(rk);
 
-        err = rd_kafka_commit0(rk, offsets, NULL,
-			       !async ? RD_KAFKA_REPLYQ(repq, 0) :
-			       RD_KAFKA_NO_REPLYQ, NULL, NULL);
+	/* Avoid conditional expression for following as that results in 0030-offset_commit
+	 * test to fail on AIX machine.
+	 */
+
+        if (!async) 
+		rq = RD_KAFKA_REPLYQ(repq, 0);
+ 
+        err = rd_kafka_commit0(rk, offsets, NULL, rq, NULL, NULL);
 
         if (!async) {
```

Comment:
I think this if .. else should be changed to:
    if (!err && !async) { 
      err = rd_kafka_q_wait..
     ..
    }